sdk/src/binary.rs

```diff
@@ -0,0 +1,283 @@
+use core::cmp::Ordering;
+
+use super::{env::internal::Env as _, xdr::ScObjectType, Env, EnvObj, EnvVal, RawVal};
+
+#[derive(Clone)]
+#[repr(transparent)]
+pub struct Binary(EnvObj);
+
+impl Eq for Binary {}
+
+impl PartialEq for Binary {
+    fn eq(&self, other: &Self) -> bool {
+        self.partial_cmp(other) == Some(Ordering::Equal)
+    }
+}
+
+impl PartialOrd for Binary {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(Ord::cmp(self, other))
+    }
+}
+
+impl Ord for Binary {
+    fn cmp(&self, other: &Self) -> core::cmp::Ordering {
+        let env = self.env();
+        let v = env.obj_cmp(self.0.to_raw(), other.0.to_raw());
+        let i = i32::try_from(v).unwrap();
+        i.cmp(&0)
+    }
+}
+
+impl TryFrom<EnvVal> for Binary {
+    type Error = ();
+
+    #[inline(always)]
+    fn try_from(ev: EnvVal) -> Result<Self, Self::Error> {
+        let obj: EnvObj = ev.try_into()?;
+        obj.try_into()
+    }
+}
+
+impl TryFrom<EnvObj> for Binary {
+    type Error = ();
+
+    #[inline(always)]
+    fn try_from(obj: EnvObj) -> Result<Self, Self::Error> {
+        if obj.as_tagged().is_obj_type(ScObjectType::Vec) {
```

Comment:
A binary is its own type.
```suggestion
        if obj.as_tagged().is_obj_type(ScObjectType::Binary) {
```

Reply:
Copy-paste error, thanks.